src/Discord.Net/WebSocket/Entities/Guilds/SocketGuild.cs

```diff
@@ -307,6 +308,7 @@ public VoiceState AddOrUpdateVoiceState(VoiceStateModel model, DataStore dataSto
                 _audioLock.Release();
             }
             await audioClient.ConnectAsync(url, CurrentUser.Id, voiceState.VoiceSessionId, token).ConfigureAwait(false);
+            _audioConnectPromise.TrySetResult(true);
```

Comment:
Use the TrySetResultAsync helper method to ensure the continuation runs on a non-gateway task.